test python 3.8, 3.10, Windows, and MacOS

sgbaird opened the pull request:
#33

sgbaird committed:
test python 3.8, 3.10, Windows, and MacOS

sgbaird committed:
python-version fix

sgbaird committed:
Update unit_tests.yml

dfm commented on .github/workflows/local_integration_tests.yml:
```diff
@@ -9,11 +9,18 @@ on:
+      matrix:
+        python:
+          - 3.8 # (almost) oldest Python supported by PSF
+          - 3.10 # newest Python that is stable
```
I think `3.10` needs to be in quotes so that it doesn't get converted to decimal.

dfm commented:
Thanks for this! One possible issue with this is that the tests are pretty expensive so I'm not sure that we want to run the full matrix. Might be sufficient to test all Pythons on Ubuntu and all platforms with one version of Python.

sgbaird committed:
python versions in quotes and exclude 3.8 for windows and macos

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations

sgbaird committed:
exclude macos and windows for Python 3.8

sgbaird commented:
@dfm sure thing! I added quotes to the Python versions (had unwittingly removed the quotes from 3.10 before - great catch). I also excluded running Python 3.8 with Windows and MacOS

sgbaird committed:
quotes around 3.8 and 3.10

https://github.com/showyourwork/showyourwork/pull/300/files#r1111068768

dfm commented:
Strangely it doesn't seem to collect _any_ tests now. I haven't had a chance to debug, but perhaps you can see what happened?

sgbaird committed:
dummy change

sgbaird committed:
Update unit_tests.yml

sgbaird committed:
Update local_integration_tests.yml

sgbaird committed:
Merge branch 'patch-1'

sgbaird committed:
rename os to platform

sgbaird commented:
@dfm I had a mistake where `os` should have been `platform` (invalid workflow file error).

sgbaird commented:
I'm running it here https://github.com/sgbaird/showyourwork/actions/runs/4212727207/jobs/7311948060

dfm commented:
OK - this is looking good! The Windows tests are failing, so I'd say let's just comment Windows out for now, and test the rest of the matrix unless you're interested in also trying to track down some of those errors in this PR too.

sgbaird committed:
comment windows for now

sgbaird committed:
comment windows for now

sgbaird commented:
Thanks! Sounds good. For provenance, here are the results from the Windows test:
```python
Run python -m pytest tests/integration
============================= test session starts =============================
platform win32 -- Python 3.10.9, pytest-[7](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:8).2.1, pluggy-1.0.0 -- C:\Miniconda\envs\test\python.exe
cachedir: .pytest_cache
rootdir: D:\a\showyourwork\showyourwork, configfile: pyproject.toml
plugins: asyncio-cooperative-0.29.0
collecting ... collected 30 items

tests/integration/test_cache.py::TestCache::test_local SKIPPED (need...) [  3%]
tests/integration/test_cache.py::TestCache::test_remote SKIPPED (nee...) [  6%]
tests/integration/test_cache.py::TestDirCache::test_local SKIPPED (n...) [ 10%]
tests/integration/test_cache.py::TestDirCache::test_remote SKIPPED (...) [ 13%]
tests/integration/test_cache_optim.py::TestLocalCacheOptimization::test_local FAILED [ 16%]
tests/integration/test_cache_optim.py::TestLocalCacheOptimization::test_remote SKIPPED [ 20%]
tests/integration/test_cache_optim.py::TestRemoteCacheOptimization::test_local SKIPPED [ 23%]
tests/integration/test_cache_optim.py::TestRemoteCacheOptimization::test_remote SKIPPED [ 26%]
tests/integration/test_default.py::TestDefault::test_local FAILED        [ 30%]
tests/integration/test_default.py::TestDefault::test_remote SKIPPED      [ 33%]
tests/integration/test_duplicate_graphics.py::TestDuplicateGraphics::test_local FAILED [ 36%]
tests/integration/test_duplicate_graphics.py::TestDuplicateGraphics::test_remote SKIPPED [ 40%]
tests/integration/test_latex.py::TestMatplotlibLatex::test_local SKIPPED [ 43%]
tests/integration/test_latex.py::TestMatplotlibLatex::test_remote SKIPPED [ 46%]
tests/integration/test_letter.py::TestLetter::test_local FAILED          [ 50%]
tests/integration/test_letter.py::TestLetter::test_remote SKIPPED (n...) [ 53%]
tests/integration/test_missing_inputs.py::TestMissingScript::test_local FAILED [ 56%]
tests/integration/test_missing_inputs.py::TestMissingScript::test_remote SKIPPED [ 60%]
tests/integration/test_missing_inputs.py::TestDeletedScript::test_local FAILED [ 63%]
tests/integration/test_missing_inputs.py::TestDeletedScript::test_remote SKIPPED [ 66%]
tests/integration/test_overleaf.py::TestOverleaf::test_local SKIPPED     [ 70%]
tests/integration/test_overleaf.py::TestOverleaf::test_remote SKIPPED    [ 73%]
tests/integration/test_pr.py::TestPullRequests::test_local SKIPPED (...) [ 76%]
tests/integration/test_pr.py::TestPullRequests::test_remote SKIPPED      [ [8](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:9)0%]
tests/integration/test_space_in_path.py::TestSpaceInPath::test_local FAILED [ 83%]
tests/integration/test_space_in_path.py::TestSpaceInPath::test_remote SKIPPED [ 86%]
tests/integration/test_variable.py::TestLatexVariable::test_local SKIPPED [ [9](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:10)0%]
tests/integration/test_variable.py::TestLatexVariable::test_remote SKIPPED [ 93%]
tests/integration/test_zenodo.py::TestZenodo::test_local SKIPPED (ne...) [ 96%]
tests/integration/test_zenodo.py::TestZenodo::test_remote SKIPPED (n...) [[10](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:11)0%]

================================== FAILURES ===================================
____________________ TestLocalCacheOptimization.test_local ____________________

self = <test_cache_optim.TestLocalCacheOptimization object at 0x000002089AD9C370>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:[11](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:12)0: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-local-cache-optimization...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-local-cache-optimization] Creating local repo `tests/sandbox/test-local-cache-optimization`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-local-cache-optimization...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-local-cache-optimization...
___________________________ TestDefault.test_local ____________________________

self = <test_default.TestDefault object at 0x000002089AD9F820>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-default...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-default] Creating local repo `tests/sandbox/test-default`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-default...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-default...
______________________ TestDuplicateGraphics.test_local _______________________

self = <test_duplicate_graphics.TestDuplicateGraphics object at 0x000002089AD9FFD0>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-duplicate-graphics...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-duplicate-graphics] Creating local repo `tests/sandbox/test-duplicate-graphics`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-duplicate-graphics...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-duplicate-graphics...
____________________________ TestLetter.test_local ____________________________

self = <test_letter.TestLetter object at 0x000002089AD[12](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:13)620>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-letter...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-letter] Creating local repo `tests/sandbox/test-letter`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-letter...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-letter...
________________________ TestMissingScript.test_local _________________________

self = <test_missing_inputs.TestMissingScript object at 0x000002089AD9FDC0>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-missing-script...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-missing-script] Creating local repo `tests/sandbox/test-missing-script`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-missing-script...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-missing-script...
________________________ TestDeletedScript.test_local _________________________

self = <test_missing_inputs.TestDeletedScript object at 0x000002089AD9EC80>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-deleted-script...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-deleted-script] Creating local repo `tests/sandbox/test-deleted-script`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-deleted-script...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-deleted-script...
_________________________ TestSpaceInPath.test_local __________________________

self = <test_space_in_path.TestSpaceInPath object at 0x000002089ADA8820>

    def test_local(self):
        """
        Test functionality by creating a new repo, customizing it,
        pushing it to GitHub, and awaiting the article build action to
        complete.
    
        """
        try:
            # Disable screen logging info from showyourwork
            self.disable_logging()
    
            # Always run this first
            self.startup()
    
            # Set up the repo
>           self.create_local()

tests\integration\helpers\temp_repo.py:295: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\integration\helpers\temp_repo.py:110: in create_local
    get_stdout(
showyourwork\subproc.py:59: in get_stdout
    return callback(code, stdout, stderr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

code = 1, stdout = ''
stderr = "error: pathspec 'first' did not match any file(s) known to git\r\nerror: pathspec 'commit'' did not match any file(s) known to git\r\n\r\nDeleting directory test-space-in-path...\r\n"

    def process_run_result(code, stdout, stderr):
        """
        Default callback function for ``get_stdout``.
    
        """
        from . import exceptions, logging
    
        # Log the output
        logger = logging.get_logger()
        if stdout:
            logger.debug(stdout)
    
        # Raise the exception
        if code != 0:
>           raise exceptions.CalledProcessError(stderr)
E           showyourwork.exceptions.other.CalledProcessError

showyourwork\subproc.py:18: CalledProcessError
---------------------------- Captured stdout call -----------------------------
[test-space-in-path] Creating local repo `tests/sandbox/test-space-in-path`...
---------------------------- Captured stderr call -----------------------------
error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-space-in-path...


------------------------------ Captured log call ------------------------------
ERROR    showyourwork:base.py:67 error: pathspec 'first' did not match any file(s) known to git

error: pathspec 'commit'' did not match any file(s) known to git



Deleting directory test-space-in-path...
============================== warnings summary ===============================
tests/integration/test_cache_optim.py::TestLocalCacheOptimization::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:9[17](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:18): PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_cache_optim.py::TestLocalCacheOptimization::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_default.py::TestDefault::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_default.py::TestDefault::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_duplicate_graphics.py::TestDuplicateGraphics::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_duplicate_graphics.py::TestDuplicateGraphics::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_letter.py::TestLetter::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_letter.py::TestLetter::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_missing_inputs.py::TestMissingScript::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_missing_inputs.py::TestMissingScript::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_missing_inputs.py::TestDeletedScript::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_missing_inputs.py::TestDeletedScript::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

tests/integration/test_space_in_path.py::TestSpaceInPath::test_local
  C:\Miniconda\envs\test\lib\site-packages\_pytest\fixtures.py:917: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
  tests/integration/test_space_in_path.py::TestSpaceInPath::test_local is using nose-specific method: `teardown(self)`
  To remove this warning, rename it to `teardown_method(self)`
  See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
    next(it)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ===========================
FAILED tests/integration/test_cache_optim.py::TestLocalCacheOptimization::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_default.py::TestDefault::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_duplicate_graphics.py::TestDuplicateGraphics::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_letter.py::TestLetter::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_missing_inputs.py::TestMissingScript::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_missing_inputs.py::TestDeletedScript::test_local - showyourwork.exceptions.other.CalledProcessError
FAILED tests/integration/test_space_in_path.py::TestSpaceInPath::test_local - showyourwork.exceptions.other.CalledProcessError
================== 7 failed, [23](https://github.com/showyourwork/showyourwork/actions/runs/4212727693/jobs/7312050330#step:5:24) skipped, 7 warnings in 9.23s ==================
Error: Process completed with exit code 1.
```

dfm commented:
Thanks!!
